Demo/Sources/YouTube/YoutubeIdentifier/Sources/YouTubeIdentifier/YouTubeIdentifier.swift

```diff
@@ -0,0 +1,23 @@
+//
+//  Copyright (c) SRG SSR. All rights reserved.
+//
+//  License information is available from the LICENSE file.
+//
+
+import Foundation
+
+/// A namespace for YouTube identifier extraction utilities.
+public enum YouTubeIdentifier {
+    /// Extracts the YouTube identifier from the given URL.
+    /// - Parameter url: The URL from which to extract the YouTube identifier.
+    /// The extracted YouTube identifier if found, or `nil` if no identifier could be extracted.
+    /// - Returns: The extracted YouTube identifier if found, or `nil` if no identifier could be extracted.
+    public static func extract(from url: URL) -> String? {
+        guard
+            let match = try? Regex("https://(?:.*youtube.*v=|.*youtu.*/(?:embed/)*(?:shorts/)*)([\\w-]+)").firstMatch(in: url.absoluteString),
```

Comment:
Instead of a single regex (which can be quite complicated to read), what do you think about the following strategy:

- Break the URL into `URLComponents`. If a `v` parameter is found then return the associated value.
- Otherwise if the host is `youtube.com` or `youtu.be` just return the last path component.

Maybe this does not cover all possible YouTube URLs but in such cases the returned id would be unknown anyway I guess.